src/libltfs/ltfs.c

```diff
@@ -4322,53 +4335,49 @@ static int _ltfs_read_rao_file(char *read_data, uint32_t *num_of_files, char *fi
 int ltfs_get_rao_list(char *path, struct ltfs_volume *vol)
 {
 	int ret = -EDEV_UNKNOWN;
+	struct rao_mod rao;
+
+	CHECK_ARG_NULL(path, -LTFS_NULL_ARG);
+	CHECK_ARG_NULL(vol, -LTFS_NULL_ARG);
+
+	memset(&rao, 0, sizeof(struct rao_mod));
+	rao.in_buf = calloc(1, RAO_MAX_RET_SIZE);
+	if (!rao.in_buf) {
+		ltfsmsg(LTFS_ERR, 10001E, "ltfs_get_rao_list: out_buf");
+		return -ENOMEM;
+	}
+
+	rao.out_buf = calloc(1, RAO_MAX_RET_SIZE);
+	if (!rao.out_buf) {
+		ltfsmsg(LTFS_ERR, 10001E, "ltfs_get_rao_list: out_buf");
+		free(rao.in_buf);
+		return -ENOMEM;
+	}
+
+	rao.buf_size = RAO_MAX_RET_SIZE;
+
 	ret = tape_device_lock(vol->device);
 	if (ret < 0) {
 		ltfsmsg(LTFS_ERR, 12010E, __FUNCTION__);
 		return ret;
 	}
 
-	/* Prepare memory and read data from file */
-	struct rao_mod rao;
-	vol->device->rao = &rao;
-	char in_buf[RAO_MAX_RET_SIZE]; /* read bytes is never larger than return size */
-	rao.in_buf = (char *)&in_buf;
-
-	if (path == NULL){
-		ret = -EDEV_INVALID_ARG;
-		goto out;
-	}
-
 	/* get rao */
-	uint32_t *tmp_num = &rao.num_of_files;
-	ret = _ltfs_read_rao_file(rao.in_buf, tmp_num, path);
+	ret = _ltfs_read_rao_file(path, rao.in_buf, RAO_MAX_RET_SIZE, &rao.in_size);
 	if (ret < 0) {
 		goto out;
 	}
 
-	/* file size check */
-	if (rao.num_of_files >= RAO_MAX_FILENUM) {
-		ltfsmsg(LTFS_ERR, 17279E, RAO_MAX_FILENUM, rao.num_of_files);
```

Comment:
`17279E` seems to been have removed and is not used anymore. Should the message be removed?

Reply:
I checked the messages by `validate_error_messages.py` and found a couple of messages to be fixed.

I made following updates

1. Change message category from ERROR to INFO because it is not a critical error from LTFS program point of view
2. Define other messages to print better information for FA